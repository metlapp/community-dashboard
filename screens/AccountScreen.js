import React, { useContext, useEffect } from "react";
import AccountModal from "../components/AccountModal";
import { Appbar, Button } from "react-native-paper";
import AuthContext from "../auth/Context";
import authStorage from "../auth/Storage";
import { SafeAreaView, StyleSheet, View } from "react-native";

const AccountScreen = () => {
  //determines what component the modal will render when a button is clicked

  const [form, setForm] = React.useState("");
  const [visible, setVisible] = React.useState(false);
  const showModal = () => setVisible(true);
  const hideModal = () => setVisible(false);
  const authContext = useContext(AuthContext);

  const handleLogout = () => {
    //Both sets the user to null and removes the user from async storage
    authContext.setUser(null);
    authStorage.removeUser();
  };
  return (
    <SafeAreaView>
      <Appbar.Header>
        <Appbar.Content title="ACCOUNT" />
      </Appbar.Header>
      {/* Opens model and takes in the visible state with what component to render */}
      <AccountModal visibility={visible} hidemodal={hideModal} form={form} />
      <View style={styles.buttonContainer}>
        {/* When Button is clicked it changes the state of the visibility and
           tells the modal what component to render inside the modal */}
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="email"
          mode="contained"
          onPress={() => {
            setForm("Email");
            setVisible(true);
          }}
        >
          Change Email address
        </Button>
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="account-edit"
          mode="contained"
          onPress={() => {
            setForm("Name");
            showModal();
          }}
        >
          Change name
        </Button>
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="square-edit-outline"
          mode="contained"
          onPress={() => {
            setForm("Password");
            showModal();
          }}
        >
          Change password
        </Button>
        <Button
          className="button"
          style={styles.buttonStyle}
          icon="square-edit-outline"
          mode="contained"
          onPress={handleLogout}
        >
          Log Out
        </Button>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  title: {
    textAlign: "center",
  },
  buttonContainer: {
    marginTop: "50%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonStyle: {
    margin: 10,
    width: "100%",
    height: 60,
    justifyContent: "center",
  },
});

export default AccountScreen;
