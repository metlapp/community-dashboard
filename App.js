import React, { useEffect } from "react";
import { StyleSheet } from "react-native";
import AuthContext from "./auth/Context";
import AccountScreen from "./screens/AccountScreen";
import { Provider as PaperProvider } from "react-native-paper";
import authStorage from "./auth/Storage";
import AuthNavigator from "./navigation/AuthNavigator";
import { NavigationContainer } from "@react-navigation/native";
import { decode, encode } from "base-64";

export default function App() {
  const [user, setUser] = React.useState();

  const restoreUser = async () => {
    const userData = await authStorage.getUser();

    //If no user data is in the storage than just return which shows the login screen
    if (!userData) {
      setUser();
      return;
    }
    setUser(userData);
  };

  //Checks the storage on open of the app
  useEffect(() => {
    restoreUser();
  }, []);

  if (!global.btoa) {
    global.btoa = encode;
  }

  if (!global.atob) {
    global.atob = decode;
  }
  return (
    <AuthContext.Provider value={{ user, setUser }}>
      <PaperProvider>
        <NavigationContainer>
          {user ? <AccountScreen /> : <AuthNavigator test={false} />}
        </NavigationContainer>
      </PaperProvider>
    </AuthContext.Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
