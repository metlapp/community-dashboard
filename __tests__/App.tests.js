import "react-native";
import React from "react";
import { shallow } from "enzyme";
import App from "../App";

// Note: this is just for use with Jest snapshot testing
// and comes packaged with react-native init project.
// You do not need this if using Enzyme 'toMatchSnapshot' etc.
import renderer from "react-test-renderer";
import AccountScreen from "../screens/AccountScreen";

// This test just uses Jest snapshot testing
it("renders correctly, test using Jest", () => {
  renderer.create(<App />);
});

// Using Jest + Enzyme
describe("<App />", () => {
  it("renders correctly, test using Jest + Enzyme", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders Child component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<AccountScreen />)).toEqual(true);
  });
});
