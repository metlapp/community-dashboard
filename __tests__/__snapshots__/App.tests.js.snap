// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransitionStart={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "rgb(242, 242, 242)",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#6200ee",
                                "elevation": 4,
                                "zIndex": 0,
                              },
                              Object {
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 3,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.24,
                                "shadowRadius": 4,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#6200ee",
                                "elevation": 0,
                                "flexDirection": "row",
                                "height": 56,
                                "marginTop": 0,
                                "paddingHorizontal": 4,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "width": 48,
                                }
                              }
                            />
                            <View
                              accessible={true}
                              focusable={false}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "flex": 1,
                                    "paddingHorizontal": 12,
                                  },
                                  Array [
                                    false,
                                    Object {
                                      "alignItems": "center",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                            >
                              <Text
                                accessibilityRole="header"
                                accessibilityTraits="header"
                                accessible={true}
                                numberOfLines={1}
                                style={
                                  Array [
                                    Object {
                                      "color": "#000000",
                                      "fontFamily": "System",
                                      "fontWeight": "400",
                                      "textAlign": "left",
                                    },
                                    Array [
                                      Object {
                                        "color": "#ffffff",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                      },
                                      Object {
                                        "fontSize": 17,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                LOGIN
                              </Text>
                            </View>
                            <View
                              style={
                                Object {
                                  "width": 48,
                                }
                              }
                            />
                          </View>
                        </RCTSafeAreaView>
                        <View
                          style={
                            Object {
                              "height": "90%",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#f2f2f2",
                                  "borderRadius": 25,
                                  "height": 55,
                                  "justifyContent": "center",
                                  "marginVertical": 15,
                                },
                                Object {
                                  "width": "100%",
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "rgb(231, 231, 231)",
                                    "borderTopLeftRadius": 4,
                                    "borderTopRightRadius": 4,
                                  },
                                  Object {},
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "rgba(0, 0, 0, 0.26)",
                                    "bottom": 0,
                                    "height": 2,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "transform": Array [
                                      Object {
                                        "scaleY": 0.5,
                                      },
                                    ],
                                  }
                                }
                              />
                              <View
                                style={
                                  Object {
                                    "minHeight": 64,
                                    "paddingBottom": 0,
                                    "paddingTop": 0,
                                  }
                                }
                              >
                                <TextInput
                                  allowFontScaling={true}
                                  autoCapitalize="none"
                                  editable={true}
                                  id="email"
                                  multiline={false}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  placeholder="Email"
                                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                                  rejectResponderTermination={true}
                                  selectionColor="#6200ee"
                                  style={
                                    Array [
                                      Object {
                                        "flexGrow": 1,
                                        "margin": 0,
                                        "textAlign": "left",
                                        "zIndex": 1,
                                      },
                                      Object {
                                        "paddingLeft": 12,
                                        "paddingRight": 12,
                                      },
                                      Object {
                                        "height": 64,
                                      },
                                      Object {
                                        "paddingBottom": 0,
                                        "paddingTop": 0,
                                      },
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 16,
                                        "fontWeight": undefined,
                                        "textAlignVertical": "center",
                                      },
                                      Array [
                                        Object {},
                                      ],
                                    ]
                                  }
                                  underlineColorAndroid="transparent"
                                  value=""
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#f2f2f2",
                                  "borderRadius": 25,
                                  "height": 55,
                                  "justifyContent": "center",
                                  "marginVertical": 15,
                                },
                                Object {
                                  "width": "100%",
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "rgb(231, 231, 231)",
                                    "borderTopLeftRadius": 4,
                                    "borderTopRightRadius": 4,
                                  },
                                  Object {},
                                ]
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "rgba(0, 0, 0, 0.26)",
                                    "bottom": 0,
                                    "height": 2,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "transform": Array [
                                      Object {
                                        "scaleY": 0.5,
                                      },
                                    ],
                                  }
                                }
                              />
                              <View
                                style={
                                  Object {
                                    "minHeight": 64,
                                    "paddingBottom": 0,
                                    "paddingTop": 0,
                                  }
                                }
                              >
                                <TextInput
                                  allowFontScaling={true}
                                  autoCapitalize="none"
                                  editable={true}
                                  id="paswword"
                                  multiline={false}
                                  onBlur={[Function]}
                                  onChangeText={[Function]}
                                  onFocus={[Function]}
                                  placeholder="Password"
                                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                                  rejectResponderTermination={true}
                                  secureTextEntry={true}
                                  selectionColor="#6200ee"
                                  style={
                                    Array [
                                      Object {
                                        "flexGrow": 1,
                                        "margin": 0,
                                        "textAlign": "left",
                                        "zIndex": 1,
                                      },
                                      Object {
                                        "paddingLeft": 12,
                                        "paddingRight": 12,
                                      },
                                      Object {
                                        "height": 64,
                                      },
                                      Object {
                                        "paddingBottom": 0,
                                        "paddingTop": 0,
                                      },
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontSize": 16,
                                        "fontWeight": undefined,
                                        "textAlignVertical": "center",
                                      },
                                      Array [
                                        Object {},
                                      ],
                                    ]
                                  }
                                  underlineColorAndroid="transparent"
                                  value=""
                                />
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#6200ee",
                                "borderColor": "transparent",
                                "borderRadius": 25,
                                "borderStyle": "solid",
                                "borderWidth": 0,
                                "elevation": 2,
                                "justifyContent": "center",
                                "marginVertical": 5,
                                "minWidth": 64,
                                "padding": 15,
                                "shadowColor": "#000000",
                                "shadowOffset": Object {
                                  "height": 0.75,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.24,
                                "shadowRadius": 1.5,
                                "width": "100%",
                              }
                            }
                          >
                            <View
                              accessibilityRole="button"
                              accessibilityState={
                                Object {
                                  "disabled": undefined,
                                }
                              }
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "overflow": "hidden",
                                  },
                                  Object {
                                    "borderRadius": 25,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  numberOfLines={1}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                      },
                                      Array [
                                        Object {
                                          "letterSpacing": 1,
                                          "marginHorizontal": 16,
                                          "marginVertical": 9,
                                          "textAlign": "center",
                                        },
                                        undefined,
                                        Object {
                                          "textTransform": "uppercase",
                                        },
                                        Object {
                                          "color": "#ffffff",
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        Object {
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "fontSize": 18,
                                        "fontWeight": "bold",
                                        "textTransform": "uppercase",
                                      }
                                    }
                                  >
                                    Login
                                  </Text>
                                </Text>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "borderColor": "transparent",
                                "borderRadius": 4,
                                "borderStyle": "solid",
                                "borderWidth": 0,
                                "elevation": 0,
                                "flexDirection": "row",
                                "justifyContent": "center",
                                "marginTop": 20,
                                "minWidth": 64,
                              }
                            }
                          >
                            <View
                              accessibilityRole="button"
                              accessibilityState={
                                Object {
                                  "disabled": undefined,
                                }
                              }
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Array [
                                  Object {
                                    "overflow": "hidden",
                                  },
                                  Object {
                                    "borderRadius": 4,
                                  },
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "justifyContent": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  numberOfLines={1}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#000000",
                                        "fontFamily": "System",
                                        "fontWeight": "400",
                                        "textAlign": "left",
                                      },
                                      Array [
                                        Object {
                                          "letterSpacing": 1,
                                          "marginHorizontal": 16,
                                          "marginVertical": 9,
                                          "textAlign": "center",
                                        },
                                        undefined,
                                        Object {
                                          "textTransform": "uppercase",
                                        },
                                        Object {
                                          "color": "blue",
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        Object {
                                          "fontFamily": "System",
                                          "fontWeight": "500",
                                        },
                                        undefined,
                                      ],
                                    ]
                                  }
                                >
                                  Forgot Password
                                </Text>
                              </View>
                            </View>
                          </View>
                          <View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000000",
                                    "fontFamily": "System",
                                    "fontWeight": "400",
                                    "textAlign": "left",
                                  },
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "System",
                                      "fontWeight": "500",
                                      "textAlign": "left",
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "justifyContent": "center",
                                        "textAlign": "center",
                                      },
                                    ],
                                    Array [
                                      Object {
                                        "fontSize": 20,
                                        "letterSpacing": 0.15,
                                        "lineHeight": 30,
                                        "marginVertical": 2,
                                      },
                                      Object {
                                        "justifyContent": "center",
                                        "textAlign": "center",
                                      },
                                    ],
                                  ],
                                ]
                              }
                            >
                              Don't have an account?
                            </Text>
                            <View
                              style={
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderColor": "transparent",
                                  "borderRadius": 4,
                                  "borderStyle": "solid",
                                  "borderWidth": 0,
                                  "elevation": 0,
                                  "minWidth": 64,
                                }
                              }
                            >
                              <View
                                accessibilityRole="button"
                                accessibilityState={
                                  Object {
                                    "disabled": undefined,
                                  }
                                }
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "overflow": "hidden",
                                    },
                                    Object {
                                      "borderRadius": 4,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#000000",
                                          "fontFamily": "System",
                                          "fontWeight": "400",
                                          "textAlign": "left",
                                        },
                                        Array [
                                          Object {
                                            "letterSpacing": 1,
                                            "marginHorizontal": 16,
                                            "marginVertical": 9,
                                            "textAlign": "center",
                                          },
                                          undefined,
                                          Object {
                                            "textTransform": "uppercase",
                                          },
                                          Object {
                                            "color": "#6200ee",
                                            "fontFamily": "System",
                                            "fontWeight": "500",
                                          },
                                          Object {
                                            "fontFamily": "System",
                                            "fontWeight": "500",
                                          },
                                          undefined,
                                        ],
                                      ]
                                    }
                                  >
                                    Register
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
    </RNCSafeAreaProvider>
  </View>
</View>
`;

exports[`<App /> renders correctly, test using Jest + Enzyme 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": undefined,
    }
  }
>
  <Provider>
    <ForwardRef(NavigationContainer)>
      <AuthNavigator
        test={false}
      />
    </ForwardRef(NavigationContainer)>
  </Provider>
</ContextProvider>
`;
